Pixel cluster tag info

mverzett opened the pull request.

vargasa committed:
DD4Hep Migration: DDPixFwdDiskAlgo

	* Migration from: Geometry/TrackerCommonData/plugins/DDPixFwdDiskAlgo.cc

vargasa committed:
Add test for DDPixFwdDiskAlgo

vargasa committed:
XML Patch from @ianna

vargasa committed:
Algo Patch from @Ianna

vargasa committed:
global to tracker namespace

henriettepetersen committed:
multi-iov implementation and condor migration for tracker-alignment validation

boudoul committed:
Adding Run 328788 (2019 MWGR1) in data relvals

zucchett committed:
First commit of the PixelClusterTagInfo class

zucchett committed:
Added PixelClusterTagInfo and Collection class info

zucchett committed:
First commit of the new PixelCluster subpackage

zucchett committed:
Added default cfg

zucchett committed:
Added readout of pixel cluster info

zucchett committed:
Updated and added parameters as number of layers

zucchett committed:
Added PixelClusterProperties struct definition

zucchett committed:
Added new int to Properties and change type to some Data

zucchett committed:
Added cluster matching; first fully working code

vargasa committed:
Use LogVerbatim instead. Expand namespaces

vargasa committed:
Test DDPixFwdDiskAlgo

vargasa committed:
Fix Variable names for Test

zucchett committed:
updated inputs and moved to dynamic allocation

zucchett committed:
Set static allocation and initialization for data struct

vargasa committed:
Fix PixFwdDiskAlgo test

vargasa committed:
Remove duplicate

zucchett committed:
Switched to one single jet collection

zucchett committed:
Added pixelCluster_cff to run on three different jet collections

zucchett committed:
Re-added include after rebase

zucchett committed:
Initialize all values to zero in struct

vargasa committed:
PosPart

zucchett committed:
Removed processes for different jet collections

zucchett committed:
Added pixel cluster tag infos to jetTools

zucchett committed:
Added pixel cluster tag infos to bTagging_cff and added fake discriminator

zucchett committed:
Added pixel cluster ro RecoBTag sequences

zucchett committed:
Added pixel cluster EventSetup_cff

zucchett committed:
Split PixelClusterTagInfoProducer in .cc and .h files

mverzett commented on PhysicsTools/PatAlgos/python/tools/jetTools.py:
```diff
@@ -273,12 +273,15 @@ def setupBTagging(process, jetSource, pfCandidates, explicitJTA, pvSource, svSou
             process.load("RecoBTag.CTagging.cTagging_cff")
             task.add(process.cTaggingTask)
+            process.load("RecoBTag.PixelCluster.pixelCluster_cff")
+            task.add(process.pixelClusterTask)
```
Questo e` potenzialmente pericoloso, rischia di rompere la backward compatibility

zucchett replied:
Ok, lo tolgo del tutto (se si puo') o c'e' un metodo migliore?

mverzett commented on DataFormats/BTauReco/interface/PixelClusterTagInfo.h:
```diff
@@ -0,0 +1,101 @@
+};
+
+struct PixelClusterData {
+    char L1_R004 = 0;
```
Curiosita`: perche` non array 2D?

zucchett replied:
Mi sembra che singoli valori siano piu' gestibili di array. Inoltre la NN che e' stata sviluppata utilizza informazioni scritte piu' o meno allo stesso modo. Magari in futuro, dopo studi ben piu' approfonditi di quelli attuali, si puo' decidere di ampliare o cambiare il modo in cui scriviamo le informazioni.

mverzett commented on RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc:
```diff
@@ -0,0 +1,357 @@
+            TVector3 j3(jetRef->px(), jetRef->py(), jetRef->pz());
+            float dR = j3.DeltaR(c3);
+            float sC = 12. * 2. / (jetRef->pt()); // * jetRef->correctedP4(0).pt()
+//            fillData(data, dR, sC, cluIt->layer, cluIt->charge);
```
Prima di fare la PR i commenti vanno rimossi

zucchett replied:
Si, certo. Ci sono anche dei cout che mi servono per capire se il codice gira. Tutto da togliere per la PR, ma si fa in pochi minuti.

mverzett commented on RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc:
```diff
@@ -0,0 +1,357 @@
+
+
+            if(cluIt->layer == 1) {
+                if(dR < 0.04) data.L1_R004++;
```
Siete consapevoli che gli anelli non sono mutualmente esclusivi ma includono quelli precedenti?

zucchett replied:
Si, sono cosi' anche nel tagger di Manuel.

mverzett commented on RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc:
```diff
@@ -0,0 +1,357 @@
+/*
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+PixelClusterTagInfoProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
```
Chiederanno di avere i descriptors settati. A me non piace, ma dura lex, sed lex

zucchett replied:
Non c'e' problema, li riaggiungo subito.

zucchett committed:
Forced embedding of pixelClusterTagInfos in slimmedJets

forthommel committed:
Using a map instead of a vector to protect for cases in which the OOT index is negative (2017 only)

forthommel committed:
Using unordered_maps

forthommel committed:
Using another unordered_map instead of a (unordered) vector

forthommel committed:
Cleanup of older pointers

forthommel committed:
Fix for clarity

forthommel committed:
Code-format applied

zucchett committed:
Removed pixelCluster from jetTools to prevent crashes

zucchett committed:
Embed pixelClusterTagInfos to patJets and keep them up to slimmedJets in miniAOD production

zucchett committed:
Cleanup code of pixelClusterTagInfoProducer class; added back fillDescriptions

Dr15Jones committed:
Modernized TrackerRecoGeometryESProducer

- Added consumes and ESGetTokens
- Added fillDescriptions
- Combined header and source files

Dr15Jones committed:
Modernized ClusterShapeHitFilterESProducer

- consolidated header and source file
- moved source to plugins directory
- added consumes and ESGetToken

Dr15Jones committed:
Added consumes calls to StripCPEESProducer

-Switched to using ESGetTokens
-Removed 3 get calls to data which were never used
-Consolidated header and source into one file

zucchett committed:
Removed unused dependencies and CMSSW version

zucchett committed:
Replaced inputTag full lists with append in miniAOD cfg

zucchett committed:
Replaced inputTag full lists with append in miniAOD cfg

Martin-Grunewald committed:
Cleanup of customisation function

cmsbuild committed:
Merge pull request #27588 from ianna/dd4hep-sim-dddworld-with-g4prodcuts-v1

DD4hep:Simulation - DDDWorld Defines Regions and Sets G4 Production Cuts

fabiocos committed:
Merge branch 'master' into sim-ddg4-production-cut-debug-v1

casarsa committed:
Switched on the CrossingFrames for BTL and ETL SimHits

ianna committed:
Update DDG4ProductionCuts.cc

a fix after megre

fabferro committed:
including ctpps in DQMOffline standard conf

zucchett committed:
Added pixelClusterTagInfos to miniAOD content

zucchett committed:
Added pixelClusterTagInfos to miniAOD content

casarsa committed:
Added MixingModule SimHits to the BTL/ETL monitoring plots; added histograms of BTL energy and time resolution, and BTL occupancy vs hit energy

casarsa committed:
Applied code-format patch

vberta committed:
no cores protection

Commit:
Enable the useful analyzer to run in the same step as simulation

vberta committed:
style fix

vberta committed:
style fix 2

Commit:
Enable plugins from the TB areas to work

tanmaymudholkar committed:
Moved ECAL validation sequence for phase 2 to more conventional location.

Dr15Jones committed:
Add consumes to ES modules in RecoLocalTracker/SiPixelRecHits

As part of the change, consolidated header and source files.

cmsbuild committed:
Merge pull request #27564 from vargasa/FailingTestXMLString

DD4Hep: DDCutTubsFromPoints Algorithm

cmsbuild committed:
Merge pull request #27597 from ianna/sim-ddg4-production-cut-debug-v1

[BugFix]  DDG4ProductionCuts Debug Printout

tanmaymudholkar committed:
Added import statement for ecalSimValid into the global validation.

Dr15Jones committed:
Added consumes calls to Phase2StripCPEESProducer

Dr15Jones committed:
Added consumes calls to ES modules in RecoLocalMuon/RPCRecHit

In addition, removed Record get calls from ES data products and moved
them into their respective ES modules.

christopheralanwest committed:
Fix for L1 EG emulation

cmsbuild committed:
Merge pull request #27583 from bsunanda/Run3-hcx212

Run3-hcx212 Bug fix for hardcoded conditions for respcorr

cmsbuild committed:
Merge pull request #27586 from forthommel/pps-diamonds-dqm-patch_11_0_X

[PPS] Diamond DQM patch for 2017

cmsbuild committed:
Merge pull request #27587 from vargasa/DD4Hep_DDPixBarLayerUpgradeAlgo_Mig

DD4Hep: Tracker Algorithm Migration DDPixBarLayerUpgrade

cmsbuild committed:
Merge pull request #27595 from cms-tsg-storm/HLTmigrationTo1100pre4

HLT migration and IrradiationBiasCorrection (110X)

cmsbuild committed:
Merge pull request #27601 from Dr15Jones/modernizeMuonTransientTrackingRecHitBuilderESProducer

Modernized MuonTransientTrackingRecHitBuilderESProducer

cmsbuild committed:
Merge pull request #27603 from Dr15Jones/consumesStripCPEESProducer

Added consumes calls to StripCPEESProducer

deguio committed:
add geometry analyzer for quick validation

vargasa committed:
Apply `code-format` patch

vargasa committed:
namespace should be trac_ following legacy convention. Modify test.

deguio committed:
raise exception in case of unknown geometry

cmsbuild committed:
Merge pull request #27598 from ianna/sim-sensitive-detector-cleanup-v1

[Sim]: Sensitive Detector Catalog Cleanup

Commit:
Correct V11 geometry for wafer types and many cfg's for correct scenarios

cmsbuild committed:
Merge pull request #27590 from bsunanda/Run3-TB38

Run3-TB38 Take care of changing geometry for HGCal DetID

cmsbuild committed:
Merge pull request #27602 from Dr15Jones/modernizeMuonDetLayerGeometryESProducer

Modernized MuonDetLayerGeometryESProducer

cmsbuild committed:
Merge pull request #27615 from Dr15Jones/consumesPhase2StripCPEESProducer

Added consumes calls to Phase2StripCPEESProducer

Commit:
Update the corrections

Commit:
Update the corrections

cmsbuild committed:
Merge pull request #27592 from civanch/g4_geom_test

Extended test on Geant4 geometry

cmsbuild committed:
Merge pull request #27618 from deguio/geomAnalyzer

add geometry analyzer for quick validation

cmsbuild committed:
Merge pull request #27604 from christopheralanwest/alca-fix-ecal-ul-ref

Update ECAL conditions to provide correct reference for 2016 and 2018 UL

cmsbuild committed:
Merge pull request #27621 from christopheralanwest/alca-hi-calo-params

Fix for L1 EG emulation in 2018 HI

Commit:
Code checks

fabferro committed:
enabling ctppsDQM for Run 2 ONLY

zhenhu committed:
replace 13TeV with 14 TeV for IB test

vargasa committed:
Fix: XMLVarNames https://github.com/cms-sw/cmssw/pull/27564#discussion_r308095587

zucchett committed:
PixelClusterTagInfoProducer updated with Code checks modifications

zucchett committed:
PixelClusterTagInfoProducer updated with Code checks modifications

ggovi committed:
Improvements to conddb tools

nancymarinelli committed:
Final (?) fix to the chi2 problem. In the end the 'source' of the different chi2 was residing in the old recovery code which has never been turned on hence the problem never spot. With this commit the recovery with the new BDT is still on. Will be switched off with the next one

nancymarinelli committed:
Code formatting

yeckang committed:
remove unnecessary line

apsallid committed:
new fragments for HGCal RelVal requests

ianna committed:
Replace PseudoTrap shape by TruncTubs

cmsbuild committed:
Merge pull request #27612 from bsunanda/Run3-sim37

Run3-sim37 Enable the useful analyzer to run in the same step as simulation

cmsbuild committed:
Merge pull request #27617 from Dr15Jones/consumesRecoLocalMuonRPCRecHit

Added consumes calls to ES modules in RecoLocalMuon/RPCRecHit

cmsbuild committed:
Merge pull request #27619 from CTPPS/ctpps_dqmoffline_cmssw_11_0

including ctpps in DQMOffline standard conf

cmsbuild committed:
Merge pull request #27607 from casarsa/mc-valMTD-update

MTD validation update

cmsbuild committed:
Merge pull request #27623 from henriettepetersen/tkal-multiIOV-condor-implementation

multi-iov implementation and condor migration for tracker-alignment v…

cmsbuild committed:
Merge pull request #27629 from ggovi/condcore-conddb-tools-improvements-7-110X

New features for the conddb tools

mmusich committed:
remove pixel_2016 modifier from 2017 era

mmusich committed:
change Global Tags to full-fledged GTs

mundim committed:
Moving to PPSPixelDigiAnalyser to TFileService; moving to LocalPoint accessor methods

jingyucms committed:
address Felice's comments

vargasa committed:
DD4hep Tracker DDPixPhase1FwdDiskAlgo Migration && test

    Geometry/TrackerCommonData/plugins/DDPixPhase1FwdDiskAlgo.cc

ianna committed:
Update 2026 scenarios with latest muon yoke description

ianna committed:
Define and use cms units

vargasa committed:
Fix logs

ianna committed:
Add 2021 cms geometry scenario

nancymarinelli committed:
And now we switch OFF the channel recovery until DPG request ot switching on for large scale tests

ianna committed:
Move Dispatchable container to Geometry where it's used

Dr15Jones committed:
Added consumes to EcalSeverityLevelESProducer

Also use ESGetToken gotten from consumes.

ianna committed:
2021 scenario with re-ordered materials and disabled algos

Dr15Jones committed:
Added consumes to EcalNextToDeadChannelESProducer

Also use ESGetToken returned by consumes.

ianna committed:
code format

schneiml committed:
Remove empty labels from METAnalyzer.

These seems to cause trouble with merging in the latest ROOT verisons.
There is no easy way of achieving the same effect ("hiding" the unused
bins), but it should not hurt much to have them numbered.

hatakeyamak committed:
Removing PFHBHERecHitCreatorMaxSample

wddgit committed:
Bug fix for setRunNumberForEachLumi

The PoolSource configuration option
setRunNumberForEachLumi had a bug that would
cause an assert failure if there was an output
module running (and possibly other problems
because the wrong run number was being used at
endRun). Probably no one was using this option
because no one noticed the failure before and there
is nothing in the repository referencing it other
than a Framework unit test. I added an output
module to the unit test so this failure would be
detected in the IB.

This bug should not affect anything if that option
is not configured. The code fragment with the
bug would not be executed.

cmsbuild committed:
Merge pull request #27622 from bsunanda/Phase2-hgx208

Phase2-hgx208 Correct V11 geometry for wafer types and many cfg's for correct scenarios

cmsbuild committed:
Merge pull request #27555 from bsunanda/Phase2-hgx205

Phase2-hgx205 Add a new scenario for HGCal V11 geometry version

cmsbuild committed:
Merge pull request #27445 from yeckang/gemDigitization

Gem digitization update

cmsbuild committed:
Merge pull request #27433 from zhenhu/add2023and2024wfs

add workflows for 2023 and 2024 scenarios

cmsbuild committed:
Merge pull request #27341 from boudoul/MWGR1

Adding Run 330098 (2019 MWGR3) in data relvals

cmsbuild committed:
Merge pull request #27600 from mmusich/phase0_pixel_charge_reweighting

Activate Pixel Charge Reweighting for phase-0 detector (2016 only)

jingyucms committed:
address Andrea's comments

hatakeyamak committed:
removing unused PFClusterProducer config files

hatakeyamak committed:
further cleanup

hatakeyamak committed:
Further cleanup

ianna committed:
address Kevin's comments

vargasa committed:
Use CMSUnits instead: See https://github.com/cms-sw/cmssw/pull/27637

ianna committed:
rebase and update D46

nancymarinelli committed:
From Silvia: Unique pointers to the tmva readers, indentation, semicolon, remove this->

emiglior committed:
bug fix in trackingMaterialGroups_ForPhaseII_D41.xml

ianna committed:
disable the unit test

nancymarinelli committed:
code formatting

smuzaffar committed:
remove GENREFLEX_ARGS flag

ianna committed:
Use correct Yoke description

ianna committed:
force to drop an external requirement

nancymarinelli committed:
Remove algo as argument from setTopology

Dr15Jones committed:
fillDescriptions now creates a cfi with the default values

The original MeasurementTrackerESProducer_cfi.py now imports the
new default cfi and uses it as the starting values.
This was the agreement with the reconstruction L2s.

cmsbuild committed:
Merge pull request #27549 from tanmaymudholkar/EcalPhase2Workflow_11_0_X

Enable Ecal DQM in Phase 2 workflows

cmsbuild committed:
Merge pull request #27550 from hatakeyamak/PF_caloFracFix

Updates to E(H)CalEnergy definitions for PF hadrons.

cmsbuild committed:
Merge pull request #27559 from CTPPS/FullSim_SimPPS-PixelDigiProducer

SimPPS/PPSPixelDigiProducer package for PPS Full simulation

cmsbuild committed:
Merge pull request #27565 from pieterdavid/stripmergedbadcomponents_removedqmstorefromesproducers

Remove SiStripBadModuleFedErrESSource

cmsbuild committed:
Merge pull request #27611 from vberta/CMSSW_11_0_0_pre3_MTV_cores_bugfix

MTV protection for samples without jets info

cmsbuild committed:
Merge pull request #27613 from bsunanda/Run3-sim38

Run3-sim38 Enable plugins from the TB areas to work

cmsbuild committed:
Merge pull request #27614 from Dr15Jones/consumesRecoLocalTrackerSiPixelRecHits

Add consumes to ES modules in RecoLocalTracker/SiPixelRecHits

cmsbuild committed:
Merge pull request #27627 from vargasa/DD4Hep_PixFwdDiskAlgo_Mig

DD4Hep: Tracker DDPixFwdDisk Algorithm

cmsbuild committed:
Merge pull request #27628 from vargasa/FixXMLVarNames

DD4Hep: Fix XML variable references

cmsbuild committed:
Merge pull request #27636 from vargasa/DD4hep_PixPhase1FwdDiskAlgo_Mig

DD4Hep: Tracker DDPixPhase1FwdDiskAlgo Migration

cmsbuild committed:
Merge pull request #27641 from ianna/sim-disp-container-move-v1

[Sim] Move the Dispatchable Container to the Geometry Package

cmsbuild committed:
Merge pull request #27643 from Dr15Jones/consumesEcalSeverityLevelESProducer

Added consumes to EcalSeverityLevelESProducer

cmsbuild committed:
Merge pull request #27645 from Dr15Jones/consumesEcalNextToDeadChannelESProducer

Added consumes to EcalNextToDeadChannelESProducer

cmsbuild committed:
Merge pull request #27649 from wddgit/bugFixSetRunNumberForEachLumi

Bug fix for setRunNumberForEachLumi

emiglior committed:
Merged TrackerPOGRecoMaterial_ForPhaseII_cmssw_11_0_X from repository emiglior with cms-merge-topic

emiglior committed:
Updates for 2023->2026 migration

ianna committed:
address Slava's comments

emiglior committed:
remove /tmp from path of output files

emiglior committed:
restore dummy version of ListGroupsMaterialDifference.h

vargasa committed:
Use LogVerbatim Instead

vargasa committed:
Remove doPos fxn. DDTran(z,x,y) -> Position(x,y,z)

ianna committed:
Move make0To2pi back to deltaPhi.h

felicepantaleo committed:
make makeHGCalValidationPlots.py executable \

yeckang committed:
revert SignalPropagationSpeed at GEMDigiProducer for consistency with comment line

ianna committed:
Move HCal algorithms to thier final destination

cmsbuild committed:
Merge pull request #27632 from ianna/dd-remove-pseudo-trap-shape-v1

DD: Replace PseudoTrap shape by TruncTubs

ianna committed:
avoid string copy

jingyucms committed:
fix some typos

ianna committed:
code format

yeckang committed:
change cspeed as cm/ns

ianna committed:
Mgrate Muon algorithms to their final destination directory

Commit:
Move all files from interface/src to plugins for TB simulation

Commit:
Code Checks

Commit:
Add provision for testing the passive material

Commit:
Correct a few includes

knash committed:
revert jets

apsallid committed:
adding new fragments to wf

cmsbuild committed:
Merge pull request #27639 from vargasa/FireworksLog

Fireworks: Change logs description

vargasa committed:
Match resize present in old algorithm

vargasa committed:
Remove duplicated entries

kpedro88 committed:
fix short matrix for 2021

cmsbuild committed:
Merge pull request #27589 from Dr15Jones/modernizeRecoTrackerMeasurementDet

Modernized ES modules in RecoTracker/MeasurementDet

cmsbuild committed:
Merge pull request #27668 from ianna/dd4hep-hcal-algo-v1

DD4hep: HCal Algorithms Moved to Geometry/HcalAlgo Package

Commit:
Update the cfg files, now that the analyzers work with simulation

ianna committed:
Radius is defined in mm

cmsbuild committed:
Merge pull request #27175 from nancymarinelli/deadChannelRecoveryBDT_V1

Ecal DPG - Deploy new software (based on BDT) to recover isolated dead  channels…

cmsbuild committed:
Merge pull request #27664 from felicepantaleo/makeHGCalValidationPlots_executable

[HGCal] make makeHGCalValidationPlots.py executable

Commit:
Use tuple

Commit:
Fix code check

christopheralanwest committed:
Updates to DT, PF hadron, HCAL and PPS calibrations for 2018 UL

makortel committed:
Add support for ESInputTag to ParameterDescription

cmsbuild committed:
Merge pull request #27637 from ianna/cms-units-v1

Define and use cms units

cmsbuild committed:
Merge pull request #27646 from ianna/dd4hep-sim-geom-builder-v2

DD4hep: Add 2021 CMS Geometry Scenario

cmsbuild committed:
Merge pull request #27669 from ianna/dd4hep-muon-algo-to-geom-v1

DD4hep: Migrate Muon Algorithms to Geometry/MuonCommonData Package

cmsbuild committed:
Merge pull request #27681 from ianna/tec-fix-units-v1

TEC: Radius is defined in mm

cmsbuild committed:
Merge pull request #27667 from yeckang/gemDigitization

revert SignalPropagationSpeed at GEMDigiProducer

cmsbuild committed:
Merge pull request #27485 from jingyucms/jingyu-lc-dev-base1100pre3

Layer Clustering for HGCal Scintillator

christopheralanwest committed:
Update pixel conditions for 2018 UL

christopheralanwest committed:
Add remaining pixel quality tags for 2018 UL MC

vargasa committed:
Apply code-format

cmsbuild committed:
Merge pull request #27594 from Dr15Jones/consumesTrackerRecoGeometryESProducer

Modernized TrackerRecoGeometryESProducer

cmsbuild committed:
Merge pull request #27596 from Dr15Jones/modernizeRecoPixelVertexingPixelLowPtUtilities

Modernized ESProducers in RecoPixelVertexing/PixelLowPtUtilities

cmsbuild committed:
Merge pull request #27670 from bsunanda/Run3-sim40

Run3-sim40 Move all files from interface/src to plugins for TB simulation

cmsbuild committed:
Merge pull request #27671 from bsunanda/Run3-sim41

Run3-sim41 Add provision for testing the passive material

cmsbuild committed:
Merge pull request #27678 from kpedro88/Run3-WF3

fix short matrix for 2021

cmsbuild committed:
Merge pull request #27683 from christopheralanwest/alca-2018UL-gt

Updates to DT, PF hadron, HCAL, PPS and pixel calibrations for 2018 UL

cmsbuild committed:
Merge pull request #27684 from makortel/parameterDescriptionESInputTag

Add support for ESInputTag to ParameterDescription

cmsbuild committed:
Merge pull request #27184 from knash/JetID2018

Include option for 2018 Jetid working points

cmsbuild committed:
Merge pull request #27661 from emiglior/TrackerPOGRecoMaterial_ForPhaseII_cmssw_11_0_X

Tracker reco material tool for phase2 in CMSSW_11_0_X

cmsbuild committed:
Merge pull request #27679 from vargasa/CMSUnits

DD4hep: Tracker usage of CMSUnits

vargasa committed:
Apply code-format

Commit:
throw a meaninful error if LFN isn't found in FileCatalog, fix ReadStreamerFile tests

Commit:
formatting

cmsbuild committed:
Merge pull request #27631 from apsallid/newfragm

[HGCal] New fragments for special HGCal RelVal requests

cmsbuild committed:
Merge pull request #27647 from schneiml/dqm-fix-metanalyzer-binlables

DQM: Remove empty labels from METAnalyzer

cmsbuild committed:
Merge pull request #27655 from smuzaffar/cleanup-genreflex-flag

remove GENREFLEX_ARGS flag

cmsbuild committed:
Merge pull request #27680 from bsunanda/Run3-TB39

Run3-TB39 Update the cfg files, now that the analyzers work with simulation

cmsbuild committed:
Merge pull request #27695 from dan131riley/streamer-noPFN-error

throw a meaninful error if LFN isn't found in FileCatalog

cmsbuild committed:
Merge pull request #27650 from hatakeyamak/Cleanup_PFHBHERecHitCreatorMaxSample_etc

Cleanup PFHBHERecHitCreatorMaxSample etc

cmsbuild committed:
Merge pull request #27689 from vargasa/OverlapsTEC

DD4Hep: Tracker overlaps TEC

zucchett committed:
Merge branch 'PixelClusterTagInfo' of https://github.com/zucchett/cmssw into PixelClusterTagInfo

zucchett committed:
Remove the embedding of all other tagInfos, as the corresponding collections are not included and crash the nanoAOD sequence

zucchett committed:
Added sequence modifier for FastSim to avoid siPixelCluster product not found

zucchett committed:
Prevent running the pixelCluster sequence when creating miniAODs from AOD produced with FastSim

zucchett committed:
Modified indentation according to CMS code-format rules

zucchett committed:
Added modifiers to exclude pixelClusterTagInfos from 80X and 94X miniAOD production

zucchett committed:
PixelClusterTagInfos switched from char to int8_t and rearranged pixel layers in arrays

zucchett committed:
Several code improvements following PR's comments

zucchett committed:
Moved .h in .cc, made producer global

zucchett committed:
Removed ifndef define endif from PixelClusterTagInfoProducer.cc

zucchett committed:
Several updates with latest comments from PR

zucchett committed:
Removed unnecessary explicit cast on input parameters

zucchett committed:
Restored charge collection as unsigned int (lost in previous commits)

zucchett committed:
Modified miniAOD sequence to access TagInfos from recoBaseTagInfosOwned_* collection

zucchett committed:
Defined dedicated accessor methods for PixelClusterTagInfo

zucchett committed:
Moved modifiers from RecoBTag_cff and partially from miniAOD_tools.py to pixelClusterTagInfos_cfi

zucchett committed:
Removed unnecessary changes on updatedPatJetsSlimmedDeepFlavour for pixelClusterTagInfo embedding

zucchett committed:
PixelClusterTagInfo code improvements and cleanup
